tools/build_steps.sh

```diff
@@ -172,20 +172,18 @@ function do_build_lib {
     echo start building
     if [ -v dynamic_list ]; then
         CFLAGS="$CFLAGS -fvisibility=protected -Wno-uninitialized" \
-        make BUFFERSIZE=20 DYNAMIC_ARCH=1 \
+        make BUFFERSIZE=20 DYNAMIC_ARCH=1 QUIET_MAKE=1 \
             USE_OPENMP=0 NUM_THREADS=64 \
             DYNAMIC_LIST="$dynamic_list" \
-            BINARY=$bitness $interface_flags $target_flags shared 2>&1 1>/dev/null
+            BINARY="$bitness" $interface_flags \
+            TARGET="$target"
     else
         CFLAGS="$CFLAGS -fvisibility=protected -Wno-uninitialized" \
-        make BUFFERSIZE=20 DYNAMIC_ARCH=1 \
+        make BUFFERSIZE=20 DYNAMIC_ARCH=1 QUIET_MAKE=1 \
             USE_OPENMP=0 NUM_THREADS=64 \
-            BINARY=$bitness $interface_flags $target_flags shared 2>&1 1>/dev/null
+            BINARY="$bitness" $interface_flags \
+            TARGET="$target"
     fi
-    echo done building, now testing
-    make BUFFERSIZE=20 DYNAMIC_ARCH=1 \
-        USE_OPENMP=0 NUM_THREADS=64 \
-        BINARY=$bitness $interface_flags $target_flags tests
```

Comment:
I think this may have been the problem. I did it to break up the build and try to get some log files, since on TravisCI there is a hard limit on the size of the logs. But `make test` requires `make shared`, and the command here was without the `DYNAMIC_LIST`. So `make` rebuilt all the kernels :(.